fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift

```diff
@@ -5,6 +5,150 @@ import IrohaCrypto
 import FearlessUtils
 
 extension AssetTransactionData {
+    static func createTransaction(
+        from item: SubscanTransferItemData,
+        address: String,
+        networkType: SNAddressType,
+        asset: WalletAsset,
+        addressFactory: SS58AddressFactoryProtocol
+    ) -> AssetTransactionData {
+        let status: AssetTransactionStatus
+
+        if item.finalized == false {
+            status = .pending
+        } else if let state = item.success {
+            status = state ? .commited : .rejected
+        } else {
+            status = .pending
+        }
+
+        let peerAddress = item.sender == address ? item.receiver : item.sender
+
+        let accountId = try? addressFactory.accountId(
+            fromAddress: peerAddress,
+            type: networkType
+        )
+
+        let peerId = accountId?.toHex() ?? peerAddress
+
+        let amount = AmountDecimal(string: item.amount) ?? AmountDecimal(value: 0)
+        let feeValue = BigUInt(item.fee) ?? BigUInt(0)
+        let feeDecimal = Decimal.fromSubstrateAmount(feeValue, precision: asset.precision) ?? .zero
+
+        let fee = AssetTransactionFee(
+            identifier: asset.identifier,
+            assetId: asset.identifier,
+            amount: AmountDecimal(value: feeDecimal),
+            context: nil
+        )
+
+        let type = item.sender == address ? TransactionType.outgoing :
+            TransactionType.incoming
+
+        return AssetTransactionData(
+            transactionId: item.hash,
+            status: status,
+            assetId: asset.identifier,
+            peerId: peerId,
+            peerFirstName: nil,
+            peerLastName: nil,
+            peerName: peerAddress,
+            details: "",
+            amount: amount,
+            fees: [fee],
+            timestamp: item.timestamp,
+            type: type.rawValue,
+            reason: nil,
+            context: nil
+        )
+    }
+
+    static func createTransaction(
+        from item: SubscanRewardItemData,
+        address: String,
+        networkType: SNAddressType,
+        asset: WalletAsset,
+        addressFactory _: SS58AddressFactoryProtocol
```

Comment:
`addressFactory` parameter is not used

Reply:
it is part of the protocol and is used by a separate implementation